File: Cpp_solving/exercise_09.cpp
/*
    Given a month and the corresponding year, report how many days the month has.
*/
#include <iostream>
using namespace std;

// I created a function to request an integer data
void request_int_data(string message, int &value){
    cout << message;
    cin >> value;
}

// Now i create the resolutive function
int how_many_days(){
    int year = 0, month = 0, days = 0;
    request_int_data("Insert a year: ", year);
    request_int_data("Insert a month: ", month);

    switch (month)
    {
    case 1:
    case 3:
    case 5:
    case 7:
    case 8:
    case 10:
    case 12:
        days = 31;
        break;
    case 4:
    case 6:
    case 9:
    case 11:
        days = 30;
        break;
    case 2:
        if ((year % 4 == 0 and year % 100 != 0) or (year % 400 == 0))
        {
            days = 29;
        }
        else days = 28;
        break;
    default:
        cout << "You didn't insert a value, try again.";
        break;
    }
    return days;
}

int main () {
    // Create a variable that contain the days
    int days = how_many_days();
    // Show the result
    cout << "The month that you inserted has " << days << " days!";
 return 0;
}

File: Python_solving/exercise_09.py
# Given a month and the corresponding year, report how many days the month has.
def is_leap_year(year):
    return year % 4 == 0 and (year % 100 != 0 or year % 400 == 0)

def how_many_days(year, month):
    days_in_month = {
        1: 31, 2: 28, 3: 31, 4: 30,
        5: 31, 6: 30, 7: 31, 8: 31,
        9: 30, 10: 31, 11: 30, 12: 31
    }
    
    if month == 2 and is_leap_year(year):
        print("This month has 29 days, because it's a leap year!")
    else:
        print(f"This month has {days_in_month.get(month, 'an invalid number of')} days!")

how_many_days(2024, 2)